test/active_record_doctor/detectors/missing_foreign_keys_test.rb

```diff
@@ -41,6 +41,20 @@ def test_uuid_missing_foreign_key_is_reported
     OUTPUT
   end
 
+  def test_destroy_async_is_not_reported
+    Context.create_table(:companies).define_model do
+      class_attribute :destroy_association_async_job, default: Class.new
```

Comment:
I understand this is normally defined when Active Job is present, but let's leave a comment stating that explicitly to avoid confusing future contributors.